src/server/scripts/EasternKingdoms/zone_searing_gorge.cpp

```diff
@@ -0,0 +1,143 @@
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Player.h"
+#include "Creature.h"
+#include "CreatureAI.h"
+#include "ScriptMgr.h"
+#include "GameObject.h"
+#include "GameObjectAI.h"
+#include "EventMap.h"
+#include "Random.h"
+#include "TemporarySummon.h"
+#include "MotionMaster.h"
+#include "WaypointDefines.h"
+#include <mutex>
+
+enum PillowFight
+{
+    NPC_SLEEPY_DARK_IRON_WORKER = 14635,
+    NPC_CHAMBERMAID_PILLACLENCHER = 14636,
+    GAMEOBJECT_DARK_IRON_PILLOW_EVENT_GENERATOR = 179829,
+
+    EVENT_WAVE_1 = 1,
+    EVENT_WAVE_2 = 2,
+    EVENT_CHAMBERMAID = 3
+};
+
+class go_dark_iron_pillow_event_generator : public GameObjectScript
+{
+public:
+    go_dark_iron_pillow_event_generator() : GameObjectScript("go_dark_iron_pillow_event_generator") {}
+
+
+    struct go_dark_iron_pillow_event_generatorAI : public GameObjectAI
+    {
+        go_dark_iron_pillow_event_generatorAI(GameObject* go) : GameObjectAI(go), owner(go) {};
+
+        void InitializeAI() override
+        {
+            events.ScheduleEvent(EVENT_WAVE_1,      60000, 0, 0);
+            events.ScheduleEvent(EVENT_WAVE_2,     180000, 0, 0);
+            events.ScheduleEvent(EVENT_CHAMBERMAID,300000, 0, 0);
+        }
+        void UpdateAI(uint32 diff) override
+        {
+            events.Update(diff);
+            while(uint32 eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_WAVE_1:
+                    for (size_t i = 0; i < 10; i++)
+                    {
+                        if (TempSummon* summon = owner->SummonCreature(NPC_SLEEPY_DARK_IRON_WORKER, -6568.03f, -1306.73f, 208.70f, 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 120000))
+                        {
+                            summon->ToCreature()->GetMotionMaster()->MovePath(NPC_SLEEPY_DARK_IRON_WORKER*100+i, true);
+                            if (i == 0)
+                                summon->ToCreature()->AI()->Talk(0);
+                        }
+                    }
+                    break;
+                case EVENT_WAVE_2:
+                    for (size_t i = 0; i < 10; i++)
+                    {
+                        if (TempSummon* summon = owner->SummonCreature(NPC_SLEEPY_DARK_IRON_WORKER, -6481.22f, -1349.06f, 212.10f, 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 120000))
+                        {
+                            summon->ToCreature()->GetMotionMaster()->MovePath(NPC_SLEEPY_DARK_IRON_WORKER * 100+10+i, true);
+
+                            if (i == 0)
+                                summon->ToCreature()->AI()->Talk(1);
+                        }
+                    }
+                    break;
+                case EVENT_CHAMBERMAID:
+                    TempSummon* summon = owner->SummonCreature(NPC_CHAMBERMAID_PILLACLENCHER, -6545.88f, -1345.29f, 208.89f,0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 300000);
```

Comment:
unused variable, needs to be commented out:
```cpp
                    TempSummon* /*summon*/ = owner->SummonCreature(NPC_CHAMBERMAID_PILLACLENCHER, -6545.88f, -1345.29f, 208.89f,0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 300000);
```
(unless you have planned to add more code using `summon` below)